src/lib.rs

```diff
@@ -196,26 +219,31 @@ where
     }
 
     /// Sets a pixel color at the given coords.
-    pub fn set_pixel(&mut self, x: u16, y: u16, color: u16) -> Result<(), ()> {
+    pub fn set_pixel(&mut self, x: u16, y: u16, color: u16) -> Result<(), Error<SE, GE>> {
         self.set_address_window(x, y, x, y)?;
         self.write_command(Instruction::RAMWR, &[])?;
         self.start_data()?;
         self.write_word(color)
     }
 
     /// Writes pixel colors sequentially into the current drawing window
-    pub fn write_pixels<P: IntoIterator<Item = u16>>(&mut self, colors: P) -> Result<(), ()> {
+    pub fn write_pixels<P: IntoIterator<Item = u16>>(
+        &mut self,
+        colors: P,
+    ) -> Result<(), Error<SE, GE>> {
         self.write_command(Instruction::RAMWR, &[])?;
         self.start_data()?;
         for color in colors {
             self.write_word(color)?;
         }
         Ok(())
     }
+
+    /// Write multiple pixels, trading ram size for speed
     pub fn write_pixels_buffered<P: IntoIterator<Item = u16>>(
         &mut self,
         colors: P,
-    ) -> Result<(), ()> {
+    ) -> Result<(), Error<SE, GE>> {
```

Comment:
I don't necessarily think one approach is better than the other, but the users could fine-tune the buffer size more if the `*_buffered` methods took the const generic parameter instead of the top-level struct. Also, users that don't want to use the buffered methods wouldn't have to specify a fake buffer size just to satisfy the API.

Reply:
Yeah I was considering that, but there were some weird dependency issues between methods calling eachother. I'll look at it more carefully and report back later.

Reply:
there's no way to add the generics to the function signature of the embedded_graphics functions afaict
![image](https://github.com/user-attachments/assets/ef989a5d-e21c-4877-bd5a-84b5bb0c119e)